Add Turbidity Unit

rgomez90 opened the pull request:
Added Turbidity Unit and conversions

* NTU
* mg/l
* ppm

Fixes #825

rgomez90 commented:
@angularsen I am having issues with the conversions tests.

I suppose I did sth wrong, but I tried to follow the wiki, by giving the conversion values a fixed value.

lipchev commented:
Hello,
I'm not at all an expert on the subject- but from what I gather- it is not really possible to relate NTU to concentrations. In fact I'm not really sure what concentration you are referring to here- as the Formazin mixture (I'm told) is made up of two different solutions. 
Do you have some other standard in mind?

angularsen commented on UnitsNet.Tests/CustomCode/TurbidityTests.cs:
```diff
@@ -0,0 +1,33 @@
+    public class TurbidityTests : TurbidityTestsBase
+    {
+        // Override properties in base class here
+        protected override double MiligramsPerLiterInOneNTU => 0.333333333;
```
How do I verify this?

If I understand correctly, turbidity needs to refer to a specific liquid in order to convert to milligrams per liter or PPM?

https://sciencing.com/definition-automated-titration-5833327.html

rgomez90 commented:
@angularsen @lipchev I think you're right. I am also not a specialist in this type of unity...But yeah, it seems, the conversion depends on a calibration. 

So I suppose, we could add NTU without conversions?

angularsen commented:
Right, the only way to support these conversions is probably to add
something like `myTurbidity.ToMilligramsPerLiter(someReference)` or create
a wrapper type `new ReferenceTurbidity(Turbidity.FromNTU(100),
someReference).ToMilligramsPerLiter();` but you lose out on much of the
generated code benefits.

We had similar issues for relative/absolute/gauge pressure conversions.

Please see ReferencePressure wrapper type introduced in
https://github.com/angularsen/UnitsNet/pull/726

rgomez90 commented:
@angularsen  I've modified the code and removed the conversions to ppm and mg/l.

I think we could merge it already just adding the new NTU unit.

For the conversion, I think we could go the `myTurbidity.ToMiligramsPerLiter(someReference)' way...

They will be just the 2 conversions, to ppm and to mg/l. So manually adding both methods and tests should not be a headache.

angularsen commented:
Ok, merged. Do you plan on adding the conversions later?

angularsen commented:
[Release UnitsNet/4.71.0 · angularsen/UnitsNet](https://github.com/angularsen/UnitsNet/releases/tag/UnitsNet%2F4.71.0)
